diff --git a/node_modules/next-auth/core/routes/callback.js b/node_modules/next-auth/core/routes/callback.js
index 04492e3..0b84ae5 100644
--- a/node_modules/next-auth/core/routes/callback.js
+++ b/node_modules/next-auth/core/routes/callback.js
@@ -129,9 +129,8 @@ async function callback(params) {
           });
           const cookieExpires = new Date();
           cookieExpires.setTime(cookieExpires.getTime() + sessionMaxAge * 1000);
-          const sessionCookies = sessionStore.chunk(newToken, {
-            expires: cookieExpires
-          });
+          const sessionChunkOptions = options.cookies.sessionToken.options.maxAge === undefined ? {} : { expires: cookieExpires };
+          const sessionCookies = sessionStore.chunk(newToken, sessionChunkOptions);
           cookies.push(...sessionCookies);
         } else {
           cookies.push({
@@ -278,9 +277,8 @@ async function callback(params) {
         });
         const cookieExpires = new Date();
         cookieExpires.setTime(cookieExpires.getTime() + sessionMaxAge * 1000);
-        const sessionCookies = sessionStore.chunk(newToken, {
-          expires: cookieExpires
-        });
+        const sessionChunkOptions = options.cookies.sessionToken.options.maxAge === undefined ? {} : { expires: cookieExpires };
+        const sessionCookies = sessionStore.chunk(newToken, sessionChunkOptions);
         cookies.push(...sessionCookies);
       } else {
         cookies.push({
@@ -396,9 +394,8 @@ async function callback(params) {
     });
     const cookieExpires = new Date();
     cookieExpires.setTime(cookieExpires.getTime() + sessionMaxAge * 1000);
-    const sessionCookies = sessionStore.chunk(newToken, {
-      expires: cookieExpires
-    });
+    const sessionChunkOptions = options.cookies.sessionToken.options.maxAge === undefined ? {} : { expires: cookieExpires };
+    const sessionCookies = sessionStore.chunk(newToken, sessionChunkOptions);
     cookies.push(...sessionCookies);
     await ((_events$signIn3 = events.signIn) === null || _events$signIn3 === void 0 ? void 0 : _events$signIn3.call(events, {
       user,
diff --git a/node_modules/next-auth/core/routes/session.js b/node_modules/next-auth/core/routes/session.js
index d23383d..45c3e02 100644
--- a/node_modules/next-auth/core/routes/session.js
+++ b/node_modules/next-auth/core/routes/session.js
@@ -75,9 +75,8 @@ async function session(params) {
         token,
         maxAge: options.session.maxAge
       });
-      const sessionCookies = sessionStore.chunk(newToken, {
-        expires: newExpires
-      });
+      const sessionChunkOptions = options.cookies.sessionToken.options.maxAge === undefined ? {} : { expires: newExpires };
+      const sessionCookies = sessionStore.chunk(newToken, sessionChunkOptions);
       (_response$cookies = response.cookies) === null || _response$cookies === void 0 || _response$cookies.push(...sessionCookies);
       await ((_events$session = events.session) === null || _events$session === void 0 ? void 0 : _events$session.call(events, {
         session: updatedSession,
